drivers: sensor: add Innovative Sensor Technology TSic xx6 driver

jeppenodgaard opened the pull request:
Add driver for TSic 206/306/316/506F/516/716 temperature sensor.
The driver uses PWM capture driver to read a single wire with Manchester-like encoding.

Commit:
drivers: sensor: add Innovative Sensor Technology TSic xx6 driver

Add driver for TSic 206/306/316/506F/516/716 temperature sensor.
The driver uses PWM capture driver to read a single wire with
Manchester-like encoding.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

MaureenHelm commented on drivers/sensor/tsic_xx6/Kconfig:
```diff
@@ -0,0 +1,11 @@
+config TSIC_XX6
+	bool "TSic xx6 driver"
+	default y
+	depends on DT_HAS_TSIC_XX6_ENABLED
```
```suggestion
	depends on DT_HAS_IST_TSIC_XX6_ENABLED
```

MaureenHelm commented on drivers/sensor/tsic_xx6/tsic_xx6.c:
```diff
@@ -0,0 +1,283 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT tsic_xx6
```
```suggestion
#define DT_DRV_COMPAT ist_tsic_xx6
```

MaureenHelm commented on drivers/sensor/tsic_xx6/tsic_xx6.c:
```diff
@@ -0,0 +1,283 @@
+		.data_bits = DT_INST_PROP(n, data_bits),                                           \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(n, &tsic_xx6_init, NULL, &tsic_xx6_data_##n, &tsic_xx6_config_##n,   \
```
`SENSOR_DEVICE_DT_INST_DEFINE`

MaureenHelm commented on dts/bindings/sensor/ist,tsic-xx6.yaml:
```diff
@@ -0,0 +1,40 @@
+  Example:
+    tsic_716: tsic_716 {
+      status = "okay";
+      compatible = "tsic-xx6";
```
```suggestion
      compatible = "ist,tsic-xx6";
```

MaureenHelm reviewed (changes requested):
Please add this new sensor to `tests/drivers/build_all/sensor` to ensure it builds in CI

Commit:
tests: drivers: build_all: sensor: add tsic_xx6

Enable PWM and PWM_CAPTURE in conf and add pwm node to app.overlay.

Add TSic xx6 node to pwm.dtsi.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

jeppenodgaard commented:
> Please add this new sensor to `tests/drivers/build_all/sensor` to ensure it builds in CI

Thanks for the review! I've applied fixes and added a commit which should enable tsic_xx6 build to CI.

MaureenHelm commented:
Needs to be rebased

jeppenodgaard commented:
@MaureenHelm and @teburd I rebased.

jeppenodgaard commented on drivers/sensor/tsic_xx6/CMakeLists.txt:
```diff
@@ -0,0 +1,2 @@
+zephyr_library()
```
Fixed.

jeppenodgaard commented on tests/drivers/build_all/sensor/pwm.dtsi:
```diff
@@ -0,0 +1,17 @@
+
```
Thanks. Fixed.
